chains/evm/contracts/ccvs/CCTPVerifier.sol

```diff
@@ -0,0 +1,528 @@
+    }
+
+    // Ensure that the message transmitter on the proxy is the same as the message transmitter on the token messenger.
+    address messageTransmitterOnProxy = address(messageTransmitterProxy.i_cctpTransmitter());
+    if (messageTransmitterOnProxy != address(messageTransmitter)) {
+      revert InvalidMessageTransmitterOnProxy(address(messageTransmitter), messageTransmitterOnProxy);
+    }
+
+    // Set the immutable state variables.
+    i_tokenMessenger = tokenMessenger;
+    i_messageTransmitterProxy = messageTransmitterProxy;
+    i_localDomainIdentifier = messageTransmitter.localDomain();
+    i_usdcToken = usdcToken;
+
+    // Approve the token messenger to burn the USDC token on behalf of this contract.
+    // The USDC token pool will be responsible for forwarding USDC it receives from the router to this contract.
+    i_usdcToken.safeIncreaseAllowance(address(i_tokenMessenger), type(uint256).max);
+
+    emit StaticConfigSet(
+      address(i_tokenMessenger), address(i_messageTransmitterProxy), address(i_usdcToken), i_localDomainIdentifier
+    );
+
+    _setDynamicConfig(dynamicConfig);
+    _setFinalityConfig(finalityConfig);
+  }
+
+  /// @inheritdoc ICrossChainVerifierV1
+  function forwardToVerifier(
+    MessageV1Codec.MessageV1 calldata message,
+    bytes32 messageId,
+    address, // feeToken
+    uint256, // feeTokenAmount
+    bytes calldata // verifierArgs
+  ) external returns (bytes memory verifierReturnData) {
+    // For EVM, sender is expected to be 20 bytes.
+    address senderAddress = address(bytes20(message.sender));
+    _assertSenderIsAllowed(message.destChainSelector, senderAddress);
+
+    Domain storage domain = s_chainToDomain[message.destChainSelector];
+    if (!domain.enabled) revert UnknownDomain(message.destChainSelector);
+
+    // We expect exactly one token transfer per message.
+    // The address of the token transferred must correspond to USDC.
+    if (message.tokenTransfer.length != 1) revert InvalidTokenTransferLength(message.tokenTransfer.length);
+    MessageV1Codec.TokenTransferV1 memory tokenTransfer = message.tokenTransfer[0];
+
+    address sourceTokenAddress = address(bytes20(tokenTransfer.sourceTokenAddress));
+    if (sourceTokenAddress != address(i_usdcToken)) revert InvalidToken(sourceTokenAddress);
+
+    if (message.tokenTransfer[0].tokenReceiver.length != 32) {
+      revert InvalidReceiver(message.tokenTransfer[0].tokenReceiver);
+    }
+
+    bytes32 decodedReceiver;
+    // For EVM chains, the mintRecipient is not used.
+    // Solana requires it, as the mintRecipient will be a PDA owned by the pool.
+    // The PDA will forward the tokens to their final destination after minting.
+    if (domain.mintRecipientOnDest != bytes32(0)) {
+      decodedReceiver = domain.mintRecipientOnDest;
+    } else {
+      decodedReceiver = abi.decode(tokenTransfer.tokenReceiver, (bytes32));
```

Comment:
This would be incompatible with the current logic that check if the pools released the correct amounts. We could modify those checks to also include the CCV calls to fix that.